gradio/cli/commands/components/_docs_utils.py

```diff
@@ -0,0 +1,747 @@
+import inspect
+import json
+import re
+import types
+import typing
+from subprocess import PIPE, Popen
+
+
+def find_first_non_return_key(some_dict):
+    """Finds the first key in a dictionary that is not "return"."""
+    for key, value in some_dict.items():
+        if key != "return":
+            return value
+    return None
+
+
+def format(code: str, type: str):
+    """Formats code using ruff."""
+    if type == "value":
+        code = f"value = {code}"
+
+    ruff_args = ["ruff", "format", "-", "--line-length=60"]
```

Comment:
Ruff may not be installed in a user's system. I think we should make this optional?

Reply:
good point. Or we could just add it as a dep? It really does make everything more readable.